src/test/ui/generator/async-recursion-forbid.rs

```diff
@@ -0,0 +1,12 @@
+// Give a clear error if async on an unboxed recursive function.
```

Comment:
This description could use some elaboration.

Reply:
Resolved by deleting the test - I found there was already one specific to this use case.